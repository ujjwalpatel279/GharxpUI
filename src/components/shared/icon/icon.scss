// ========================================================================
// COMPONENTS/SHARED/ICON
// ========================================================================

@import '~sassHelpers';

.c-icon {
  display: inline-block;
  width: 1em;
  height: 1em;
  stroke-width: 0;
  fill: color(grey3);
  stroke: currentColor;

  &--primary {
    fill: var(--primary);
    &__info {
      path:first-of-type {
        fill: color(grey3);
      }
      path:last-of-type {
        fill: color(grey1);
      }
    }
  }

  &--blue {
    fill: color(blue);
  }

  &--small {
    font-size: font-size(2);
  }

  &--large {
    font-size: font-size(3);
  }
}
