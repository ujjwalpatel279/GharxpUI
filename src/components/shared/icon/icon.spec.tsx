import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Icon } from './icon';

describe('Container Component', () => {
  afterEach(cleanup);

  it('renders an icon', () => {
    const { container } = render(<Icon name="close" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders an icon with the primary colour', () => {
    const { container } = render(<Icon name="close" colour="primary" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders an icon with the secondary colour', () => {
    const { container } = render(<Icon name="close" colour="secondary" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders a small icon', () => {
    const { container } = render(<Icon name="close" size="small" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders a large icon', () => {
    const { container } = render(<Icon name="close" size="large" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
