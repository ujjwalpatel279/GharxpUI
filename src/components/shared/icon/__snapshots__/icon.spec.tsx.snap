// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders a large icon 1`] = `
<svg
  class="c-icon c-icon--large"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M0.939 0.939c0.586-0.586 1.536-0.586 2.121 0l4.939 4.939 4.939-4.939c0.586-0.586 1.536-0.586 2.121 0s0.586 1.536 0 2.121l-4.939 4.939 4.939 4.939c0.586 0.586 0.586 1.536 0 2.121s-1.536 0.586-2.121 0l-4.939-4.939-4.939 4.939c-0.586 0.586-1.536 0.586-2.121 0s-0.586-1.536 0-2.121l4.939-4.939-4.939-4.939c-0.586-0.586-0.586-1.536 0-2.121z"
  />
</svg>
`;

exports[`Container Component renders a small icon 1`] = `
<svg
  class="c-icon c-icon--small"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M0.939 0.939c0.586-0.586 1.536-0.586 2.121 0l4.939 4.939 4.939-4.939c0.586-0.586 1.536-0.586 2.121 0s0.586 1.536 0 2.121l-4.939 4.939 4.939 4.939c0.586 0.586 0.586 1.536 0 2.121s-1.536 0.586-2.121 0l-4.939-4.939-4.939 4.939c-0.586 0.586-1.536 0.586-2.121 0s-0.586-1.536 0-2.121l4.939-4.939-4.939-4.939c-0.586-0.586-0.586-1.536 0-2.121z"
  />
</svg>
`;

exports[`Container Component renders an icon 1`] = `
<svg
  class="c-icon"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M0.939 0.939c0.586-0.586 1.536-0.586 2.121 0l4.939 4.939 4.939-4.939c0.586-0.586 1.536-0.586 2.121 0s0.586 1.536 0 2.121l-4.939 4.939 4.939 4.939c0.586 0.586 0.586 1.536 0 2.121s-1.536 0.586-2.121 0l-4.939-4.939-4.939 4.939c-0.586 0.586-1.536 0.586-2.121 0s-0.586-1.536 0-2.121l4.939-4.939-4.939-4.939c-0.586-0.586-0.586-1.536 0-2.121z"
  />
</svg>
`;

exports[`Container Component renders an icon with the primary colour 1`] = `
<svg
  class="c-icon c-icon--primary__close"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M0.939 0.939c0.586-0.586 1.536-0.586 2.121 0l4.939 4.939 4.939-4.939c0.586-0.586 1.536-0.586 2.121 0s0.586 1.536 0 2.121l-4.939 4.939 4.939 4.939c0.586 0.586 0.586 1.536 0 2.121s-1.536 0.586-2.121 0l-4.939-4.939-4.939 4.939c-0.586 0.586-1.536 0.586-2.121 0s-0.586-1.536 0-2.121l4.939-4.939-4.939-4.939c-0.586-0.586-0.586-1.536 0-2.121z"
  />
</svg>
`;

exports[`Container Component renders an icon with the secondary colour 1`] = `
<svg
  class="c-icon c-icon--secondary__close"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M0.939 0.939c0.586-0.586 1.536-0.586 2.121 0l4.939 4.939 4.939-4.939c0.586-0.586 1.536-0.586 2.121 0s0.586 1.536 0 2.121l-4.939 4.939 4.939 4.939c0.586 0.586 0.586 1.536 0 2.121s-1.536 0.586-2.121 0l-4.939-4.939-4.939 4.939c-0.586 0.586-1.536 0.586-2.121 0s-0.586-1.536 0-2.121l4.939-4.939-4.939-4.939c-0.586-0.586-0.586-1.536 0-2.121z"
  />
</svg>
`;
